### DISCUSSION TITLE
TrinoHook inconsistency with  connections in secrets.

### DISCUSSION THREAD

**Discussion by Juacall** [apache/airflow#26012] [Q&A]
### Apache Airflow Provider(s)

trino

### Apache Airflow version

2.2.3 (latest released)

### Operating System

macOS Monterey

### Deployment

Virtualenv installation

### Deployment details

Basic deployment with the providers trino and mysql. Not using fernet key.

### What happened

The Trino hook requires a username and password to use the hook. If i were to try to use a connection without it, it would error out asking for basic authentication or a trino user. If i add a username and password to a connection. The connection becomes encrypted and is added to secrets then i won't be able to discover it unless i go through the basehook.

conn = BaseHook.get_connection(kwargs['my_conn_id'])

### What you expected to happen

I'm allowed to use the hook without a username and password and or I can add a connection i retrieved from a base hook to the trino hook.

### How to reproduce

Run a trino server without any authentication. Use the Trino hook with the trino_default connection. 
Will receive error Basic Authentication or X-Trino-User must be sent.

Add basic authentication to Trino server and to airflow trino_default  connection. Try use TrinoHook. 
Will receive error. conn_id trino_default not defined.

### Anything else

There are flaws in both systems that causes the inconsistency  with the TrinoHook when using Basic Authentication. This will allow a connection from secrets to be passed to TrinoHook but will not fix the problem with TrinoHook Basic Authentication. Basic Authentication will not work with a Http connection and Airflow does not allow you to send a https connection without using a uri.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by wolfier** [comment]
Please update the title to reflect the nature of your issue

**Comment by eladkal** [comment]
Feel free to submit a PR

**Comment by Juacall** [comment]
> Feel free to submit a PR

How do i test changes?

**Comment by potiuk** [comment]
> How do i test changes?

https://github.com/apache/airflow/blob/main/CONTRIBUTORS_QUICK_START.rst

**Comment by eladkal** [comment]
@alexandermalyga since you work with Trino maybe you can assist on this issue?

**Comment by alexandermalyga** [comment]
> @alexandermalyga since you work with Trino maybe you can assist on this issue?

I haven't been able to reproduce the issue (or I'm not understanding it correctly).
When running Trino without authentication, it's enough to just provide some username without a password.
When running with authentication, `TrinoHook` picks up the provided connection (or the default `trino_default`) just fine.

Tried both options using Airflow 2.3.4 and `apache-airflow-providers-trino==4.0.0` without any issues.

**Comment by eladkal** [comment]
Thank you for checking. Converting this issue to discussion as it seems to be something local
